dask_kubernetes/core.py

```diff
@@ -14,20 +17,207 @@
+                self._pod.metadata.name, self.namespace
+            )
+        await super().close(**kwargs)
+
+    async def logs(self):
+        try:
+            log = await self.core_api.read_namespaced_pod_log(
+                self._pod.metadata.name, self.namespace
+            )
+        except ApiException as e:
+            if "waiting to start" in str(e):
+                log = ""
+            else:
+                raise e
+        return Log(log)
+
+    async def describe_pod(self):
+        self._pod = await self.core_api.read_namespaced_pod(
+            self._pod.metadata.name, self.namespace
+        )
+        return self._pod
+
+    def __repr__(self):
+        return "<Pod %s: status=%s>" % (type(self).__name__, self.status)
+
+
+class Worker(Pod):
+    """ A Remote Dask Worker controled by SSH
+    Parameters
+    ----------
+    scheduler: str
+        The address of the scheduler
+    address: str
+        The hostname where we should run this worker
+    connect_kwargs: dict
+        kwargs to be passed to asyncssh connections
+    kwargs:
+        TODO Document Worker kwargs
+    """
+
+    def __init__(self, scheduler: str, **kwargs):
+        super().__init__(**kwargs)
+
+        self.scheduler = scheduler
+
+        self.pod_template.metadata.labels["dask.org/component"] = "worker"
+        self.pod_template.spec.containers[0].env.append(
+            kubernetes.client.V1EnvVar(
+                name="DASK_SCHEDULER_ADDRESS", value=self.scheduler
+            )
+        )
+
+
+class Scheduler(Pod):
+    """ A Remote Dask Scheduler controled by SSH
+    Parameters
+    ----------
+    address: str
+        The hostname where we should run this worker
+    connect_kwargs: dict
+        kwargs to be passed to asyncssh connections
+    kwargs:
+        TODO Document Scheduler kwargs
+    """
+
+    def __init__(self, scheduler_timeout: str, **kwargs):
+        super().__init__(**kwargs)
+        self.service = None
+        self._scheduler_timeout = scheduler_timeout
+
+        self.pod_template.metadata.labels["dask.org/component"] = "scheduler"
+        self.pod_template.spec.containers[0].args = [
+            "dask-scheduler",
+            "--idle-timeout",
+            self._scheduler_timeout,
+        ]
+
+    async def start(self, **kwargs):
+        await super().start(**kwargs)
+
+        while (await self.describe_pod()).status.phase == "Pending":
+            await asyncio.sleep(0.1)
+
+        while self.address is None:
+            logs = await self.logs()
+            for line in logs.splitlines():
+                if "Scheduler at:" in line:
+                    self.address = line.split("Scheduler at:")[1].strip()
+            await asyncio.sleep(0.1)
+
+        self.service = await self._create_service()
+        self.address = "tcp://{name}.{namespace}:{port}".format(
+            name=self.service.metadata.name, namespace=self.namespace, port=8786
```

Comment:
Is this port intended to be hard-coded?  Or does it not matter because of Kubernetes magic?

Reply:
Yeah it is kubernetes magic. The scheduler runs on the default ports and we need to tell kubernetes what those are. I agree that having magic numbers floating about is unpleasant. 

Would it be preferable to move to a constant?